File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.cpp
#include <string>
#include <vector>

using namespace std;

string base_n(int num,int n) {
    string s="";
    while(num!=0) {
        s=(num%n<10)?(to_string(num%n)+s):((char)(55+num%n)+s);
        num/=n;
    }
    return s;
}

string solution(int n, int t, int m, int p) {
    string r="",st="0";
    int num=1;
    for(int i=0;i<t*m;i++) {
        st+=base_n(num++,n);
    }
    for(int i=p-1;i<t*m;i+=m) {
        r+=st[i];
    }
    return r;
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def solution(n, t, m, p):
    r,st,num='','0',1
    for i in range(t*m):
        st+=base_n(num,n)
        num+=1
    for i in range(p-1,t*m,m):
        r+=st[i]
    return r

def base_n(num, n):
    s=''
    while num!=0:
        s=str(num%n)+s if num%n<10 else chr(55+num%n)+s
        num//=n
    return s